docs/RFD-JobWorkerService.md

```diff
@@ -0,0 +1,366 @@
+  This is configured by setting [memory.max](https://docs.kernel.org/admin-guide/cgroup-v2.html). 
+  
+  - _`IOBytesPerSecond`_ - maximum read and write on the device mounted / is mounted on. 
+  This is configured by setting [io.max](https://docs.kernel.org/admin-guide/cgroup-v2.html) in the cgroup for the process. 
+  For example, a IOBytesPerSecond of 1000000000 would be similar to `259:1 rbps=1000000000 wbps=1000000000 riops=max wiops=max` in the cgroup's `io.max` file. 
+
+  - _`Command`_ - command to execute. For example, `_/bin/bash_`. 
+  
+  - _`Arguments`_ - arguments to pass to the command. 
+  For example, `"-c" "echo hello"` would be provided to command and ultimately run similar to `/bin/bash -c "echo hello".`
+
+
+* **func New(config JobConfig) \*Job** - create new Job instance, based on provided job configuration.
+
+
+* **func (\*Job) Start() error** - start job executions and return [error](https://pkg.go.dev/errors) if job failed to start or required fields contains no value.
+
+  Starting new job tart takes following steps:
+
+  - create new [cgroup](https://docs.kernel.org/admin-guide/cgroup-v2.html) for the job - e.g `/sys/fs/cgroup/<UUID>`
+  - create subtree to control resources - e.g. `+cpu +memory +io` into file `/sys/fs/cgroup/<UUID>/cgroup.subtree_control`
+  - set value for CPU, Memory and IO limits based on job configuration 
+    - `CPU` - set by writing QUOTA/PERIOD into `/sys/fs/cgroup/<UUID>/tasks/cpu.max`
+    - `Memory` - set by writing the provided IOBytesPerSecond to `/sys/fs/cgroup/<uuid>/tasks/memory.max`
+    - `IO` - is set by writing IOBytesPerSecond into `<MAJOR NUMBER OF PHYSICAL DEVICE>:<MINOR NUMBER OF PHYSICAL DEVICE> rbps=<IOBytesPerSecond> wbps=<IOBytesPerSecond> riops=max wiops=max` to `/sys/fs/cgroup/<UUID>/tasks/io.max` <br/><br/> _Note_:  `<MAJOR NUMBER OF PHYSICAL DEVICE>` and `<MINOR NUMBER OF PHYSICAL DEVICE>` shoudl be dynamically detected and provided during runtime.
+  - configures a new [exec.Cmd](https://pkg.go.dev/os/exec#Cmd)
+    - sets clone flags for creating new mount, pid, and network namespaces 
+    - sets [unshare flags](https://pkg.go.dev/syscall#SysProcAttr) for mount so that new mounts are not reflected in host mount namespace 
+    - sets [UseCgroupFD and CgroupFD](https://pkg.go.dev/syscall#SysProcAttr) to the cgroup's file descriptor such as `/sys/fs/cgroup/<UUID>/tasks`
+  - runs the configured `exec.Cmd` in a new goroutine
+    - does not wait for the command to complete 
+    - the goroutine will stop when the process completes or stopped
+
+
+* **func (\*Job) Status() JobStatus** - returns the current status of the job (see [JobStatus](#type-JobStatus))
+
+
+* **func (\*Job) Stream() OutputReader** - returns an `OutputReader` (type that implements [io.Reader](https://pkg.go.dev/io#Reader)). 
+`OutputReader` return entire output from when job started to the current output. `Stream` can call when job started, executing or completed.
```

Comment:
Could you elaborate bit more about: 
How a client stream part will be notified about new data produced by a process and where data will be stored (memory/file) ?

Reply:
> and where data will be stored (memory/file) ?

When I've clarified details with @r0mant  - he mentioned no data persistence and all should keep in memory.  But initial though was to keep:
- keep out put in memory
- flush on disk (file) periodically or on completion/stopping job. 
-

Reply:
I agree, keeping the data in memory is totally acceptable. Please update the documentation to reflect this.

Reply:
![Screenshot 2024-08-29 at 11 49 39 AM](https://github.com/user-attachments/assets/ca3efb52-37ba-41e5-9f08-a8f9ef155822)